### PR TITLE
Add support for declaring specific endpoints to be built 

### LINKED ISSUE #46: Watch/build a specific entry point
### The feature

Say, for example, that I have three entry points in a project:
- `frontend.js`
- `editor.js`
- `admin.js`

If I'm working on a frontend module, I likely will not need to rebuild `admin.js` or `editor.js` when making changes, as their scopes are separate from the code I'm working on. Therefore, watching and building their respective trees is unnecessary overhead in terms of both time and system resources.
This is especially true of gutenberg-related code which, in my experience, is where the most significant proportion of time is spent on a build task. On one of my projects (that isn't currently using build-tools), removing the gutenberg entry point from the Webpack build cuts the build time down by approx 66%.

It would be great if I could pass a parameter to the watch/build task to declare only the entry point(s) I want to target when I'm working on a specific feature; e.g. `--entry=frontend`.

### PR DESCRIPTION
**Pull request by con322**
## Description
This PR adds the ability to define the endpoints you want to build so that if your working on a specific part of a project you can optimise performance/speed of builds for what your working on via defining the endpoints after `@` symbol. For example for all projects you can use `@frontend` or if your working with multiple project and only want to build the frontend assets for one you can use `test-plugin@frontend`. Fixes #46

### Change log
* Add logic so that `projects` positional can have entrypoints defined  after them by appened `@` along with the entrypoint ie `test-plugin@frontend` or multiple via `+` such as `test-plugin@frontend+editor`.
* Adds new property to project config called `filteredEntrypoints ` which will contain the specified entrypoints if set, defaults to empty array. 
* Add cli test.
* Add unit test for new util function (`getFilteredEntryPoints`).

## Code reviewer(s)
* Is `filteredEntrypoints` the right variable name? 
* Should some kind of validation be performed to ensure projects/filtered endpoints are following the correct format (`plugin@fronted+editor`, `@frontend`, etc)? 
* `README.md` will be updated once an initial review has been done and the above questions have been answered as things are subject to change.

## Testing
Example Individual Projects build command: `build-tools build my-plugin@frontend`.
Example Site-wide build command: `build-tools build @frontend`.

* Verify Individual Projects build command works without `@` specified
* Verify Individual Projects build command works with single `@frontend` specified
* Verify Individual Projects build command works with multiple `@frontend+editor` specified
* Verify Site-wide build command works without `@` specified
* Verify Site-wide build command works with single `@frontend` specified
* Verify Site-wide build command works with multiple `@frontend+editor` specified

### CODE REVIEW

**Review by jaymcp — APPROVED** [review]
Thanks Connor! Works for me 😊

File: src/utils/get-filtered-entrypoints.js
Comment (ampersarnie) [inline src/utils/get-filtered-entrypoints.js:1]: Can you add a docblock to this please.

File: src/utils/entrypoints.js
Comment (ampersarnie) [inline src/utils/entrypoints.js:23]: Because you're doing the same thing twice, here and on L32, you can probably move this outside of the `if()` statement on L22 and reuse `type` for L32.

**Review by ampersarnie — CHANGES_REQUESTED** [review]
I've not tested yet and only reviewed code. Looks great, I've left some minor feedback for you. Also, can you update the docs in `docs/` with how this works to provide guidance to users.

File: example-site/yarn.lock
Comment (ampersarnie) [inline example-site/yarn.lock:1]: Just spotted this, yarn isn't used in the project so can this be removed to avoid confusion.

File: yarn.lock
Comment (ampersarnie) [inline yarn.lock:1]: Just spotted this, yarn isn't used in the project so can this be removed to avoid confusion.

**Review by ampersarnie — CHANGES_REQUESTED** [review]
Good to go after these are removed.

File: example-site/yarn.lock
Comment (con322) [reply example-site/yarn.lock:1]: Removed this :)

File: yarn.lock
Comment (con322) [reply yarn.lock:1]: Removed this :)

**Review by ampersarnie — APPROVED** [review]

File: src/commands/build.js
Comment (ampersarnie) [inline src/commands/build.js:63]: The order of these is somewhat incorrect. Mapping last will cause a project target of only entry points to return an array with a single empty string. If we've not defined any projects we need to check this before filtering or it will assume `@entrypoint` is the project.

A simple swap of logic order works.

```suggestion
const projectsList = projects.split(',').map((item) => item.split('@')[0]).filter((item) => item.length > 0);
```

File: src/commands/build.js
Comment (ampersarnie) [inline src/commands/build.js:65]: I found that `@entrypoint` can be treat by this logic as a project. So with an entrypoint only set in the comment, this will return a false-positive. Seems like we can get around this by adding another logical operator against `projectsList` length.

```suggestion
  const isAllProjects = (site || hasTargetDirs) && (!projects || projectsList.length == 0);
```

There may be a cleaner, less convoluted way of doing this by my quick test allowed me to build entrypoints only.

**Review by ampersarnie — CHANGES_REQUESTED** [review]
Sorry I was just doing one last check before I merged this and came across an issue with the building all entry points feature you documented.

Testing steps taken:

**Single entrypoint**
```bash
cd ./example-site
node ./../src/cli.js build @editor
```

**Multiple entrypoints**
```bash
cd ./example-site
node ./../src/cli.js build @editor+frontend
```

File: src/commands/build.js
Comment (con322) [reply src/commands/build.js]: Made this change in my latest, commit. Wanted to test locally.

File: src/commands/build.js
Comment (con322) [reply src/commands/build.js]: Made this change in my latest, commit. Wanted to test locally.

**Review by ampersarnie — APPROVED** [review]

**Pull request merged**
